app/views/bills/_form.html.erb

```diff
@@ -22,11 +34,13 @@
     <%= f.text_field :usage, :id => "usage" %>
   </div>
 
-  <%= f.fields_for :units do |units_form| %>
-    <div class="field">
-      <%= units_form.label :number_occupants, "Number of Occupants in Your Home or Apartment" %><br>
-      <%= units_form.text_field :number_occupants, :id => "occupants" %>
-    </div>
+  <% if @show_occupants %>
```

Comment:
I think overall it isn't great to have to have that logic in the controller twice (both in `create` and `new`) as it is less DRY. Furthermore Sandi Metz suggests that more than one instance ("@") variable in the controller is not great (see also https://github.com/thoughtbot/guides/tree/master/best-practices#rails). To shorten it I wonder do we need to check for a signed_in user at all? Can it be sufficient to simply check for current_user.unit.present?. The documentation for .try might also yield a useful pattern here: http://apidock.com/rails/v4.2.1/Object/try.

Anyways some of this is merely trade-offs so I don't want to hold this up, but wanted to provide some helpful patterns to think about/try. Let me know if these suggestions make sense but if not happy to take it as is.